src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -193,7 +193,7 @@ public static void ImplementsDictionary_DictionaryOfString()
             }
 
             {
-                Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<StringToStringIReadOnlyDictionaryWrapper>(JsonString));
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<StringToStringIReadOnlyDictionaryWrapper>(JsonString));
```

Comment:
What does the entire exception look like now?

Reply:
```
 System.NotSupportedException : The collection type 'System.Text.Json.Serialization.Tests.StringToStringIReadOnlyDictionaryWrapper' is not supported.
      Stack Trace:
        ...\JsonPropertyInfoCommon.cs(193,0): at System.Text.Json.JsonPropertyInfoCommon`4.CreateDerivedDictionaryInstance(JsonPropertyInfo collectionPropertyInfo, IDictionary sourceDictionary, String jsonPath, JsonSerializerOptions options)
...
```